src/chunk_index.c

```diff
@@ -712,3 +754,85 @@ chunk_index_mark_clustered(Oid chunkrelid, Oid indexrelid)
 	CommandCounterIncrement();
 	heap_close(rel, AccessShareLock);
 }
+
+PG_FUNCTION_INFO_V1(chunk_index_recreate_create);
+Datum
+chunk_index_recreate_create(PG_FUNCTION_ARGS)
+{
+	Oid			chunk_index_oid = PG_GETARG_OID(0);
+	Relation	chunk_index_rel;
+	Relation	hypertable_index_rel;
+	Relation	hypertable_rel;
+	Relation	chunk_rel;
+	Oid			constraint_oid;
+	Oid			new_chunk_indexrelid;
+	Chunk *chunk;
+	ChunkIndexMapping *cim; 
+
+	chunk_index_rel = relation_open(chunk_index_oid, AccessShareLock);
+
+	chunk = chunk_get_by_relid(chunk_index_rel->rd_index->indrelid, 0, true);
+	cim = chunk_index_get_by_indexrelid(chunk, chunk_index_oid);
+
+	hypertable_rel = relation_open(cim->hypertableoid, AccessShareLock);
```

Comment:
Should probably do `heap_open()` here as it has extra checks that this is indeed a heap relation as opposed to index. (fwiw, I haven't been consistent with this either).

Reply:
done (and all similar below too)